Packet++/header/NflogLayer.h

```diff
@@ -0,0 +1,289 @@
+		uint8_t* getRecordBasePtr() const { return (uint8_t*)m_Data; }
+
+		/**
+		 * @return A pointer to the value of the record as byte array (the 'V' in Type-Length- __Value__)
+		 */
+		uint8_t* getValue() const { return m_Data->recordValue; }
+	};
+
+	/**
+	 * @class NflogLayer
+	 * Represents an NFLOG protocol layer
+	 */
+	class NflogLayer : public Layer
+	{
+	public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to ether_header)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		NflogLayer(uint8_t* data, size_t dataLen, Packet* packet) : Layer(data, dataLen, NULL, packet) { m_Protocol = NFLOG; }
+
+        /**
+		 * A constructor that allocates a new Nflog header with empty fields
+		 */
+		NflogLayer();
+
+		~NflogLayer() {}
+
+		/**
+		 * Get a pointer to the Nflog header.
+		 * @return A pointer to the nflog_header
+		 */
+		nflog_header* getNflogHeader() const { return (nflog_header*)m_Data; }
+
+        /**
+         * Set address family in header of the packet.
+         * @param[in] family as uint8_t
+        */
+        void setFamily(uint8_t family);
+
+		/**
+		 * Get address family of the packet. e.g. 2 for ipv4 and 10 for ipv6
+		 * @return an unsigned char of address family
+		*/
+		uint8_t getFamily();
+
+        /**
+         * Set version number in header of the packet.
+         * @param[in] version as uint8_t
+        */
+        void setVersion(uint8_t version);
+
+        /**
+         * Get Version number inside packet header
+         * The version field is 0 for the current version of the pseudo-header
+         * @return an unsigned char for version
+        */
+        uint8_t getVersion();
+
+        /**
+         * Set resource id in header of the packet.
+         * @param[in] resourceId as uint16_t
+        */
+        void setResourceId(uint16_t resourceId);
+
+        /**
+         * Get Resource Id in packet header
+         * On one netlink socket it's possible to listen to several nflog groups; the resource ID is the nflog group for the packet
+        */
+        uint16_t getResourceId();
+
+		/**
+		 * get packet header of the packet as the tlv value of 1
+		 * @return pointer to nflog_packet_header
+		*/
+		nflog_packet_header* getPacketHeader();
+
+		/**
+		 * returns a pair of pointer to tlv data and the length of the tlv
+		 * @param[in] type type of tlv by using enum class defined as NflogTlvType
+		 * @return NflogTlv obtained by type
+		*/
+		NflogTlv getTlvByType(NflogTlvType type) const;
+
+		// implement abstract methods
+
+		/**
+		 * Currently identifies the following next layers: IPv4Layer, IPv6Layer using address family
+		 * Otherwise sets PayloadLayer
+		 */
+		void parseNextLayer();
+
+		/**
+		 * @return Size of nflog_header
+		 */
+		size_t getHeaderLen() const { return sizeof(nflog_header); }
```

Comment:
The header len should be `sizeof(nflog_header)` + the size of all TLV records besides the data (payload) of `NflogTlvType::NFULA_PAYLOAD`.

There should be some logic that goes over the TLV records and sums their length until it reaches `NflogTlvType::NFULA_PAYLOAD` and then adds only the first 4 bytes of it (type + length)

Reply:
I thought only sizeof(nflog_header) would be sufficient. I changed this